src/FtlStream.cpp

```diff
@@ -23,75 +23,74 @@
 #pragma region Constructor/Destructor
 FtlStream::FtlStream(
     std::shared_ptr<FtlControlConnection> controlConnection,
-    std::unique_ptr<ConnectionTransport> mediaTransport,
-    const MediaMetadata mediaMetadata,
-    const ftl_stream_id_t streamId,
-    const ClosedCallback onClosed,
-    const RtpPacketCallback onRtpPacket,
-    const bool nackLostPackets)
+    ftl_stream_id_t streamId,
+    const ClosedCallback onClosed)
 :
     controlConnection(std::move(controlConnection)),
-    mediaTransport(std::move(mediaTransport)),
-    mediaMetadata(mediaMetadata),
     streamId(streamId),
-    onClosed(onClosed),
-    onRtpPacket(onRtpPacket)
+    onClosed(onClosed)
 {
-    // Prepare stream data stores to accept packets from SSRCs specified by control handshake
-    ssrcData.try_emplace(mediaMetadata.AudioSsrc);
-    ssrcData.try_emplace(mediaMetadata.VideoSsrc);
-
     // Bind to FtlStream
     this->controlConnection->SetFtlStream(this);
-
-    // Bind to media transport callbacks
-    this->mediaTransport->SetOnBytesReceived(
-        std::bind(
-            &FtlStream::mediaBytesReceived,
-            this,
-            std::placeholders::_1));
-    this->mediaTransport->SetOnConnectionClosed(std::bind(&FtlStream::mediaConnectionClosed, this));
 }
 #pragma endregion
 
 #pragma region Public methods
-Result<void> FtlStream::StartAsync(uint16_t mediaPort)
+Result<void> FtlStream::StartMediaConnection(
+        std::unique_ptr<ConnectionTransport> mediaTransport,
+        const uint16_t mediaPort,
+        const MediaMetadata mediaMetadata,
+        const FtlMediaConnection::RtpPacketCallback onRtpPacket)
 {
-    std::unique_lock lock(dataMutex);
-    // Fire up our media connection
-    Result<void> mediaStartResult = mediaTransport->StartAsync();
-    if (mediaStartResult.IsError)
+    std::scoped_lock lock(mutex);
```

Comment:
Why `scoped_lock` instead of `lock_guard` in a lot of these places? `scoped_lock` is new to me but I was under the impression that it was more useful for locking multiple mutexes.

Reply:
That's true! It can lock multiple, but it also locks a single mutex just as efficiently as lock_guard. So the consensus seems might as well use it everywhere for consistency.

https://rules.sonarsource.com/cpp/RSPEC-5997